### PR TITLE
feat(query): top k syntax support

### LINKED ISSUE #14972: Feature: top k syntax support
**Summary**

Description for this feature.

example:
```
select TOP 4 c1 from testable ORDER BY c1;

-- similar to 

select c1 from testable order by c1 limit 4;


```

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

### Parameters
n

The maximum number of rows to return in the result set.

### Example 

```sql
select TOP 4 c1 from testable ORDER BY c1;

-- similar to

select c1 from testable order by c1 limit 4;

```
- Fixes #14972 

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [x] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15318)

### CODE REVIEW

**Comment by TCeason** [comment]
cc @soyeric128

**Comment by andylokandy** [comment]
what if `select top from t1` where top is a column in t1?

**Review by andylokandy — APPROVED** [review]

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -2602,6 +2602,15 @@ pub fn hint(i: Input) -> IResult<Hint> {
...
 }
 
+pub fn top_n(i: Input) -> IResult<u64> {
+    map(
+        rule! {
+            TOP ~ ^#literal_u64
+        },
+        |(_, n)| n,
+    )(i)
+}
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:2612]:
```suggestion
pub fn top_n(i: Input) -> IResult<u64> {
    map_res(
        rule! {
            TOP ~ #literal_u64? : "TOP <n>"
        },
        |(_, opt_n)| {
            if let Some(n) = opt_n {
                Ok(n)
            } else {
                Err(nom::Err::Failure(ErrorKind::Other(
                    "expecting a literal number after keyword `TOP`, if you were refering a column with name `top`, \
                    please quote it like `\"top\"`"
                )))
            }
        },
    )(i)
}
```

Comment (andylokandy) [reply src/query/ast/src/parser/statement.rs:2612]: Also add `select top from t1` to `test_statement_error()` in `src/query/ast/tests/it/parser.rs`

Comment (BohuTANG) [reply src/query/ast/src/parser/statement.rs]: Before merging I missing this comemnts, cc @TCeason

Comment (TCeason) [reply src/query/ast/src/parser/statement.rs]: > Before merging, I missed these comments. cc @TCeason

It's ok, Andy and I push for this in commit:

https://github.com/datafuselabs/databend/pull/15318/commits/1c7cda284c43ab64d431bc96acc2c4d44b27a324
https://github.com/datafuselabs/databend/pull/15318/commits/db3cf37dfceb334cbf2fa27dceebf291d9ea4d30

**Pull request merged**